osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/datadecoder/BasicDlmsDataDecoderTest.java

```diff
@@ -0,0 +1,101 @@
+// SPDX-FileCopyrightText: Copyright Contributors to the GXF project
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.opensmartgridplatform.adapter.protocol.dlms.domain.datadecoder;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.openmuc.jdlms.datatypes.CosemDate;
+import org.openmuc.jdlms.datatypes.CosemDateTime;
+import org.openmuc.jdlms.datatypes.CosemDateTime.ClockStatus;
+import org.openmuc.jdlms.datatypes.CosemTime;
+import org.openmuc.jdlms.datatypes.DataObject;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.utils.DlmsHelper;
+import org.opensmartgridplatform.adapter.protocol.dlms.exceptions.ProtocolAdapterException;
+
+class BasicDlmsDataDecoderTest {
+  private final DlmsHelper dlmsHelper = new DlmsHelper();
+  private final BasicDlmsDataDecoder decoder = new BasicDlmsDataDecoder(this.dlmsHelper);
+
+  @ParameterizedTest
+  @EnumSource(
+      value = DataObject.Type.class,
+      names = {
+        "INTEGER",
+        "LONG_INTEGER",
+        "DOUBLE_LONG",
+        "LONG64",
+        "UNSIGNED",
+        "LONG_UNSIGNED",
+        "DOUBLE_LONG_UNSIGNED",
+        "LONG64_UNSIGNED"
+      })
```

Comment:
This list is not complete when conserning numbers. Missing   ENUMERATE,     FLOAT32,    FLOAT64

Reply:
The list contains all integer numbers, floats require a different decoding (and are not yet used in the protocol adapter). And for enums we should add separate decoders, as each enum has a different meaning.